activesupport/test/core_ext/module/attribute_accessor_per_thread_test.rb

```diff
@@ -2,53 +2,56 @@
 require 'active_support/core_ext/module/attribute_accessors_per_thread'
 
 class ModuleAttributeAccessorPerThreadTest < ActiveSupport::TestCase
-  def setup
-    @class = Class.new do
-      thread_mattr_accessor :foo
-      thread_mattr_accessor :bar,  instance_writer: false
-      thread_mattr_reader   :shaq, instance_reader: false
-      thread_mattr_accessor :camp, instance_accessor: false
-    end
+  class Foo
```

Comment:
What @matthewd pointed out sounds reasonable. If it didn't work properly before then we should be good to raise. Would be nice to have the inline classes in the test-case work though. Maybe using `def name; "MyClass" end` does the trick.